### ISSUE TITLE
kvs: reconsider use of EPERM in lookup/commit API

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#1228]
in the internal KVS lookup & commit APIs, the errno EPERM is used when returning an error to user for an internally inconsistent state.  For example, if a directory reference points to a non-directory tree object.

This is perhaps not the best errno to describe this and should be reconsidered to something else.

**Comment by chu11** [comment]
thinking about this a bit, EFAULT might be a good replacement.  For anything related to a bad blobref or invalid references, this makes perfect sense, as the blobref is a "bad address".

There may not be any errno that is perfect for anything that is just "internally inconsistent".  ENOTRECOVERABLE seems like a pretty good one.

**Comment by chu11** [comment]
Perhaps EUCLEAN "structure needs cleaning", it's not bad and sort of similar to the idea of "file system corruption".

**Comment by garlick** [comment]
I kind of like ENOTRECOVERABLE (State not recoverable).  EFAULT and EPERM are both too common to me and tend to point to specific problems that might throw someone chasing an issue off course.

If I saw ENOTRECOVERABLE pop up, I'd probably think "WTF?", then grep source.  If I saw EPERM or EFAULT I might jump to conclusions about security or bad pointers and follow a false lead.

For reference here is a list of POSIX.2008 error numbers and descriptions

http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/errno.h.html
